### PR TITLE
Adding support for notebook python code with markup notebook-python

### PR DESCRIPTION
**Pull request by pravakarg**
With this change, now codes from Jupyter notebook can be referenced with markup 
[!notebook-python[](codefiles/create-workspace.ipynb?name=writeConfig)] . 

The codes from jupyter node book from a cell with metadata tag with name  "writeConfig" will be added to document.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4989)

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs [hunk 1]
```diff
@@ -188,6 +191,47 @@ protected override void Write(HtmlRenderer renderer, CodeSnippet codeSnippet)
...
+            }
+            catch (JsonException)
+            {
+                _context.LogError("mutiple-tags-with-same-name", string.Format("The name '{0}' is not unique in the notebook file.", tagName), obj);
+                return string.Empty;
+            }
+
+            if (sourceObject == null)
+            {
+                _context.LogError("source-code-notfound", string.Format("The name '{0}' is not present in the notebook file.", tagName), obj);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: `source-code-notfound` => `source-code-not-found` for consistency with existing codes

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Microsoft.DocAsCode.MarkdigEngine.Extensions.csproj [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
   <ItemGroup>
     <PackageReference Include="Markdig" Version="0.16.0" />
+	<PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Microsoft.DocAsCode.MarkdigEngine.Extensions.csproj]: minor: change tab to 4 spaces

**Comment by superyyrrzz** [comment]
Could you also add some test case in https://github.com/dotnet/docfx/blob/dev/test/Microsoft.DocAsCode.MarkdigEngine.Tests/CodeSnippetTest.cs ?

**Comment by superyyrrzz** [comment]
@yishengjin1413 Please help review this markdig change.

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs [hunk 3]
```diff
@@ -188,6 +191,47 @@ protected override void Write(HtmlRenderer renderer, CodeSnippet codeSnippet)
...
+            string sourceJsonPath = string.Format("$..cells[?(@.metadata.name=='{0}')].source", tagName);
+            JToken sourceObject = null;
+            try
+            {
+                sourceObject = contentObject.SelectToken(sourceJsonPath);
+            }
+            catch (JsonException)
+            {
+                _context.LogError("mutiple-tags-with-same-name", string.Format("The name '{0}' is not unique in the notebook file.", tagName), obj);
+                return string.Empty;
```
Comment (yishengjin1413) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: @cillroy Would you give some suggestions about the error messages?

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs (hunk 1 above)
Comment (pravakarg) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: As per my understanding, if source file is not found, source-code-not-found is used.  It is about tag not found. Changed accordingly.

**Comment by yufeih** [comment]
@yishengjin1413 @superyyrrzz  do you have guys have other comments about this PR? LGTM.

**Pull request merged**